### DISCUSSION TITLE
Shipstation Integration not Including Shipping Method

### DISCUSSION THREAD

**Discussion by coderreco** [swellstores/community#569] [Help] [answered]
My client is using Shipstation for their fulfillment. We have activated the official Shipstation integration which does successfully post orders, but it does not show the selected order method anywhere in the order on the Shipstation side. 

All orders are successfully selecting a shipping method when required(physical items in cart), but not showing anywhere on Shipstation. Example:
```
Order from console:
  "shipping": {
    // sensitive data omitted
    "service_name": "Local Pickup",
    "service": "local_pickup"
  },
```

Same order in Shipstation:
<img width="437" alt="image" src="https://github.com/user-attachments/assets/d624f0cf-70a0-4d6a-804b-be8ead224c4d">
The expected behaviour is to see something related to the chosen shipping service in the `Requested` field.

I have submitted this to support with screenshots but have not yet been given any information on where the disconnect is. 

Has anyone else faced this issue or something similar? Is there a workaround I can use such as also adding the shipping service to order notes or metadata in order to show it in notes on the Shipstation time?

Thanks in advance.

**Comment by coderreco** [comment]
Answer received, gap found in ShipStation integration where requestedShippingService is not being assigned when orders are created. Was told support is working on it so closing this discussion for tidiness.

Edit: Reopening this because no fix has been implemented

**Reply by logeshswell** [reply]
@coderreco Our team is working on it, and it will be shipped this week.

**Reply by coderreco** [reply]
Good to hear! Thanks @logeshswell

**Reply by logeshswell** [reply, marked as answer]
The fix is rolled out to production. Can you check and confirm if it's working fine for new orders?

**Reply by coderreco** [reply]
Looks like it is working correctly @logeshswell, thanks! One other gap in this integration is it does not pass any variant information to ShipStation. Passed this along through support but will share here with an example:
The first item in this order is a variant:
![Screenshot 2024-09-12 at 10 03 03 AM](https://github.com/user-attachments/assets/5070bb9f-1dcb-4bd3-87e9-d232a04948c5)

From console:
```
"items": [
    {
      "product_id": "64436847c857ff00124c5890",
      "quantity": 1,
      "options": [
        {
          "id": "645590e67cf3e19da7f69055",
          "value": "Dry",
          "name": "Option",
          "variant": true,
          "value_id": "645590fc7cf3e19da7f69056"
        }
      ],
      "id": "66dfb369ac4941001213c5a5",
      "variant_id": "645590fc850fdf0013191f82",
      "price": 160,
      "orig_price": 160,
      "delivery": "shipment",
      "shipment_weight": 0,
      "product_name": "Teoxane Advanced Filler",
      // ...rest of fields
    },
      // ...rest of items
  ]
  // ...rest of cart
}
```

In my client's use case this is an issue for the distribution team. Although there is SKU information visible in screenshot that is just leftover data that was necessary for a since-cancelled integration. Won't be added for new products and therefore isn't usable.

I ended up making a custom integration instead so this isn't urgent for me, but I believe this should be updated for future users. Here is what it looks like when including the options:
![Screenshot 2024-09-12 at 10 07 10 AM](https://github.com/user-attachments/assets/8c5944ba-c530-41a7-ae24-20b73f50b4d3)

What are the chances of Swell open-sourcing their pre-built integrations so that we may contribute to their improvement? According to [this post](https://github.com/orgs/swellstores/discussions/171#discussioncomment-5246164) from @ericingram in March 2023 the integrations were built quickly with limited scope and there is not time to iterate. Also specifically mentions "**Integrations we build in-house will be open-source, so you'll be able to add new functionality like this easily.**". I would be happy to contribute to these integrations to make them more complete in coverage.

**Reply by ericingram** [reply]
> What are the chances of Swell open-sourcing their pre-built integrations so that we may contribute to their improvement? According to [this post](https://github.com/orgs/swellstores/discussions/171#discussioncomment-5246164) from @ericingram in March 2023 the integrations were built quickly with limited scope and there is not time to iterate. Also specifically mentions "Integrations we build in-house will be open-source, so you'll be able to add new functionality like this easily.". I would be happy to contribute to these integrations to make them more complete in coverage.

We do plan to open source the integrations once we make Swell Apps generally available. All of the existing integrations will be turned into open source apps that you can contribute to, or fork and install privately.

**Reply by logeshswell** [reply]
> One other gap in this integration is it does not pass any variant information to ShipStation.

Our team is working on it and will keep you posted if I get some updates.

**Reply by coderreco** [reply]
> > One other gap in this integration is it does not pass any variant information to ShipStation.
> 
> Our team is working on it and will keep you posted if I get some updates.

Thanks @logeshswell
